src/main/java/com/n1303/birthdaybuddy/repository/FriendshipRepository.java

```diff
@@ -61,4 +61,20 @@ public List<User> getAllFriendsOfUser(String userId) {
 
         return userRepository.getAllUsersInRange(friendsUserIdList);
     }
+
+    public boolean delete(String firstUser, String secondUser) {
+        Optional<Friendship> existingFriends = friendshipMap.values().stream().filter(friendship -> {
+            return Objects.equals(friendship.getFirstUserId() , firstUser) &&
+                    Objects.equals(friendship.getSecondUserId() , secondUser) ||
+                    Objects.equals(friendship.getFirstUserId() , secondUser) &&
+                            Objects.equals(friendship.getSecondUserId() , firstUser);
+        }).findAny();
+
+        if (existingFriends.isPresent()) {
+            friendshipMap.remove(existingFriends.get().getFriendshipId());
```

Comment:
I think you can do
`return friendShipMap.remove(...`
 it gives a boolean itself